rllib/offline/estimators/tests/test_dm.py

```diff
@@ -0,0 +1,250 @@
+import unittest
+from ray.rllib.evaluation.worker_set import WorkerSet
+from ray.rllib.execution.rollout_ops import synchronous_parallel_sample
+from ray.rllib.algorithms import AlgorithmConfig
+
+import numpy as np
+from ray.rllib.offline.estimators import DirectMethod
+from ray.rllib.offline.estimators.tests.gridworld import GridWorldPolicy, GridWorldEnv
+from ray.rllib.policy.policy import Policy
+from ray.rllib.policy.sample_batch import SampleBatch, concat_samples
+from ray.rllib.utils.framework import try_import_torch
+
+import ray
+
+_, nn = try_import_torch()
+
+
+class TestDM(unittest.TestCase):
+    """Learning tests for the DirectMethod Estimator"""
+
+    @classmethod
+    def setUpClass(cls):
+        ray.init()
+        # Epsilon-greedy exploration values
+        random_eps = 0.8
+        mixed_eps = 0.5
+        expert_eps = 0.2
+        cls.num_episodes = 32
+        cls.gamma = 0.99
+        num_workers = 8
+
+        # Config settings for FQE model used in DM estimator
+        cls.q_model_config = {
+            "n_iters": 600,
+            "minibatch_size": 32,
+            "tau": 1.0,
+            "model": {
+                "fcnet_hiddens": [],
+                "activation": "linear",
+            },
+            "lr": 0.01,
+        }
+
+        env = GridWorldEnv()
+        cls.random_policy = GridWorldPolicy(
+            observation_space=env.observation_space,
+            action_space=env.action_space,
+            config={"epsilon": random_eps},
+        )
+        cls.mixed_policy = GridWorldPolicy(
+            observation_space=env.observation_space,
+            action_space=env.action_space,
+            config={"epsilon": mixed_eps},
+        )
+        cls.expert_policy = GridWorldPolicy(
+            observation_space=env.observation_space,
+            action_space=env.action_space,
+            config={"epsilon": expert_eps},
+        )
+        del env
```

Comment:
why do you explicitly need to del env?

Reply:
it will be gced once it's out of scope, right?

Reply:
Fixed